LTSketchbook/Part Number/6000/6603/DC1304A_B/DC1304A_B.ino

```diff
@@ -0,0 +1,363 @@
+However, the Linduino is only possible because of the Arduino team's commitment
+to the open-source community.  Please, visit http://www.arduino.cc and
+http://store.arduino.cc , and consider a purchase that will help fund their
+ongoing work.
+ */
+
+/*! @file
+    @ingroup DC1304
+*/
+
+#include <Arduino.h>
+#include <stdint.h>
+#include "Linduino.h"
+#include "LT_SPI.h"
+#include "LT_I2C.h"
+#include "UserInterface.h"
+#include "QuikEval_EEPROM.h"
+#include "LTC6603.h"
+#include <SPI.h>
+#include <Wire.h>
+
+// Function Declaration
+void print_title();                         // Print the title block
+void print_prompt();                        // Prompt the user for an input command
+void print_user_command(uint8_t menu);      // Display selected differential channels
+void menu_1_filter_settings();              // Sets the gain and cutoff frequencies
+void menu_2_set_gpo();                      // Sets the GPO
+void menu_3_shutdown();                     // Shuts down the LTC6603
+void menu_4_poweron();                      // Power up the LTC6603
+
+// Global variables
+uint8_t spi_out;
+uint8_t control_byte;
+uint8_t filter_gain_settings = LTC6603_GAIN_0dB;
+uint8_t filter_lp_settings = LTC6603_LPF0 ;
+uint8_t filter_noFunc_settings = LTC6603_NOFUNC0 ;
+uint8_t gpio_settings = LTC6603_GPO_LOW ;
+uint8_t shdn_settings = LTC6603_SHDN;
+float freq_settings = LTC6603_FREQ_MIN; 
+
+//! Initialize Linduino
+void setup()
+{
+  quikeval_SPI_init();  // Configure the spi port for 4MHz SCK
+  Serial.begin(115200); // Initialize the serial port to the PC
+  print_title();        // Displays the title
+  print_prompt();       // Display user options
+  control_byte = 0x00;  // A global variable that contains the whole control byte.
+
+}
+
+//! Repeats Linduino loop
+void loop()
+{
+  int16_t user_command;         // The user input command
+  if (Serial.available())       // Check for user input
+  {
+    user_command = read_int();        // Read the user command
+    if (user_command != 'm')
+      Serial.println(user_command);   // Prints the user command to com port
+
+    switch (user_command)
+    {
+      case 1:
+        menu_1_filter_settings();
+        break;
+      case 2:
+        menu_2_set_gpo();
+        break;
+      case 3:
+        menu_3_shutdown();
+        break;
+      case 4:
+        menu_4_poweron();
+        break;
+      case 5:
+        menu_5_changeclockfrequency();
+        break;
+      default:
+        Serial.println("Incorrect Option");
+        break;
+    }
+    print_prompt();
+  }
+}
+
+// Function Definitions
+
+//! Set filter settings
+//! @return void
+void  menu_1_filter_settings()
+{
+  int16_t user_command; // The user input command
+  while (1)
+  {
+    Serial.print(F("*************************\n\n"));
+    Serial.println (F("0-Enter External Frequency"));
+    Serial.println (F("1-Set Gain"));
+    Serial.println (F("2-Set Lowpass"));
+    Serial.println (F("3-Set No Function Code"));
```

Comment:
Can these menu options be printed from the print_prompt(). If not, print_prompt() can be removed.

Reply:
It's ok.  this is different from the general print_prompt.  This is only for menu item 1.